Arm/Memory/SeparateAutomation.lean

```diff
@@ -308,19 +361,36 @@ def consumeRewriteFuel : SimpMemM Unit :=
 def outofRewriteFuel? : SimpMemM Bool := do
   return (← get).rewriteFuel == 0
 
+/-- Create a trace note that folds `header` with `(NOTE: can be large)`,
+and prints `msg` under such a trace node.
+-/
+def SimpMemM.traceLargeMsg (header : MessageData) (msg : MessageData) : SimpMemM Unit :=
+    withTraceNode m!"{header} (NOTE: can be large)" do
+      trace[simp_mem.info] msg
+
+
 def getConfig : SimpMemM SimpMemConfig := do
   let ctx ← read
   return ctx.cfg
 
 /-
-Introduce a new definition into the local context,
+Introduce a new definition into the local context, simplify it using `simp`,
 and return the FVarId of the new definition in the goal.
 -/
-def introDef (name : String) (hdefVal : Expr) : SimpMemM FVarId  := do
+def simpAndIntroDef (name : String) (hdefVal : Expr) : SimpMemM FVarId  := do
   SimpMemM.withMainContext do
+
     let name ← mkFreshUserName <| .mkSimple name
     let goal ← getMainGoal
     let hdefTy ← inferType hdefVal
+
+    /- Simp to gain some more juice out of the defn.. -/
+    let (simpCtx, simprocs) ← LNSymSimpContext
```

Comment:
This is a heavy `simp`, what with `state_simp_rules`, `bitvec_rules`, and `minimal_theory` in there. Any pitfalls?

Reply:
@bollu ?

Reply:
bitvec_rules, minimal_theory, `state_value`.

Reply:
Addressed!